### PR TITLE
Implement Google civic API backend

### PR DESCRIPTION
**Pull request by juanuribe28**
Add civic servlet to query information about the representatives that are in charge of the user's division.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/CivicServlet.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    } finally {
+      Map<String, String> apiQueryParams = new HashMap<>();
+      apiQueryParams.put("key", API_KEY);
+      apiQueryParams.put("address", address);
+      String jsonResult = UrlRequest.urlQuery(API_PATH, apiQueryParams);
+      response.setContentType("application/json;");
+      response.getWriter().println(jsonResult);
+    }
+  }
+}
```
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/CivicServlet.java:59]: [Nit] add newline

File: src/main/java/com/google/sps/servlets/CivicServlet.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Map<String, String> locationQueryParams = new HashMap<>();
+    locationQueryParams.put("lat", latitude);
+    locationQueryParams.put("lng", longitude);
+    String locationUrl = String.format("%s://%s:%s/location",
+        request.getScheme(), 
+        request.getServerName(), 
+        request.getServerPort());
+    String address = "";
+    try {
+      // Location URL doesn't work on devserver, so isntead use hardcoded string.
```
Comment (dranilew) [inline src/main/java/com/google/sps/servlets/CivicServlet.java]: [Nit] "Instead" spelled incorrectly.

File: src/main/java/com/google/sps/servlets/CivicServlet.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      Type mapType = new TypeToken<Map<String, String>>() {}.getType();
+      Map<String, String> locationMap = gson.fromJson(locationJsonResult, mapType);
+      address = String.format("%s %s, %s, %s %s", 
+          locationMap.get("Street Number"),
+          locationMap.get("Street Name"),
+          locationMap.get("City"),
+          locationMap.get("State"),
+          locationMap.get("Zip Code"));
+    } catch (FileNotFoundException fnfe) {
+      System.out.println(fnfe);
```
Comment (dranilew) [inline src/main/java/com/google/sps/servlets/CivicServlet.java]: [Optional] You can alternatively use `fnfe.printStackTrace();` instead.

**Review by dranilew — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/CivicServlet.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      Type mapType = new TypeToken<Map<String, String>>() {}.getType();
+      Map<String, String> locationMap = gson.fromJson(locationJsonResult, mapType);
+      address = String.format("%s %s, %s, %s %s", 
+          locationMap.get("Street Number"),
+          locationMap.get("Street Name"),
+          locationMap.get("City"),
+          locationMap.get("State"),
+          locationMap.get("Zip Code"));
+    } catch (FileNotFoundException fnfe) {
+      System.out.println(fnfe.printStackTrace());
```
Comment (dranilew) [inline src/main/java/com/google/sps/servlets/CivicServlet.java]: [Nit] .printStackTrace() already prints the stack trace to System.out, so `System.out.println()` is unnecessary.

**Review by dranilew — CHANGES_REQUESTED** [review]

**Review by dranilew — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/CivicServlet.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    Map<String, String> locationQueryParams = new HashMap<>();
+    locationQueryParams.put("lat", latitude);
+    locationQueryParams.put("lng", longitude);
+    String locationUrl = String.format("%s://%s:%s/location",
+        request.getScheme(), 
+        request.getServerName(), 
+        request.getServerPort());
+    String address = "";
+    try {
+      // Location URL doesn't work on devserver, so instead use hardcoded string.
```
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/CivicServlet.java]: [Nit] I think this comment makes more sense if you put it in the catch block below.

**Review by daperlmut — CHANGES_REQUESTED** [review]

**Review by daperlmut — APPROVED** [review]

**Pull request merged**